src/node_watchdog.cc

```diff
@@ -13,6 +13,10 @@ Watchdog::Watchdog(v8::Isolate* isolate, uint64_t ms) : isolate_(isolate),
   loop_ = new uv_loop_t;
   CHECK(loop_);
   rc = uv_loop_init(loop_);
+  if (rc != 0) {
+       FatalError("node::Watchdog::Watchdog()",
+               "Failed to initialize loop. Maybe the file limit is reached?");
```

Comment:
Yeah, I think @addaleax is right that it may be difficult to do x-platform reliably. The message could be extended a bit with something like, `Failed to initialize loop. This may be caused, for instance, by reaching the file limit.`